wallet/src/de/schildbach/wallet/ui/dashpay/PlatformRepo.kt

```diff
@@ -239,6 +250,51 @@ class PlatformRepo(val walletApplication: WalletApplication) {
         }
     }
 
+    /**
+     *  Wraps callbacks of DeriveKeyTask as Coroutine
+     */
+    private suspend fun deriveKey(handler: Handler, wallet: Wallet, password: String): KeyParameter {
+        return suspendCoroutine { continuation ->
+            object : DeriveKeyTask(handler, walletApplication.scryptIterationsTarget()) {
+
+                override fun onSuccess(encryptionKey: KeyParameter, wasChanged: Boolean) {
+                    continuation.resume(encryptionKey)
+                }
+
+                override fun onFailure(ex: KeyCrypterException?) {
+                    //CreateIdentityService.log.error("unable to decrypt wallet", ex)
+                    continuation.resumeWithException(ex as Throwable)
+                }
+
+            }.deriveKey(wallet, password)
+        }
+    }
```

Comment:
This does not appear to be used, but having this method in here could help remove a lot of callback type methods.

Reply:
One thing I wonder is if there is an exception, what gets executed after this:  `continuation.resumeWithException`